js/src/hooks/useIsMounted.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback, useRef } from '@wordpress/element';
+
+/**
+ * Returns a function to check whether the caller component is mounted or unmounted.
+ * Usually, it's used to avoid the warning - Can't perform a React state update on an unmounted component.
+ *
+ * @return {() => boolean} A function returns a boolean indicates the caller component is mounted or unmounted.
```

Comment:
:nail_care: 
I'm not born with English/English native, but Grammarly suggests:
```suggestion
 * @return {() => boolean} A function returns a boolean that indicates the caller component is mounted or unmounted.
```